src/java-interop/Directory.Build.targets

```diff
@@ -0,0 +1,101 @@
+<Project>
+
+  <Import Project="..\..\build-tools\scripts\NativeToolchain.targets" />
+
+  <PropertyGroup>
+    <_JavaInteropLibName Condition=" '$(OS)' != 'Windows_NT' And Exists ('/Library/Frameworks/') ">libjava-interop.dylib</_JavaInteropLibName>
+    <_JavaInteropLibName Condition=" '$(OS)' != 'Windows_NT' And !Exists ('/Library/Frameworks/') ">libjava-interop.so</_JavaInteropLibName>
+    <_JavaInteropLibName Condition=" '$(OS)' == 'Windows_NT' ">java-interop.dll</_JavaInteropLibName>
```

Comment:
We have a lot of existing code using `$(OS)`, but I think the proper check is to use these:

```
$([MSBuild]::IsOSPlatform ('osx'))
$([MSBuild]::IsOSPlatform ('windows'))
$([MSBuild]::IsOSPlatform ('linux'))
```

I've seen DevCom issues where someone had `%OS%` set to `Windows` on their Dell PC!